d2/src/main/java/com/linkedin/d2/balancer/dualread/DualReadLoadBalancer.java

```diff
@@ -95,28 +95,50 @@ public DualReadLoadBalancer(LoadBalancerWithFacilities oldLb, LoadBalancerWithFa
   public void start(Callback<None> callback)
   {
     // Prefetch the global dual read mode
-    _dualReadStateManager.checkAndSwitchMode(null);
+    DualReadModeProvider.DualReadMode mode = _dualReadStateManager.getGlobalDualReadMode();
 
-    _newLb.start(new Callback<None>()
-    {
+    // if in new-lb-only mode, new lb needs to start successfully to call the callback. Otherwise, the old lb does.
+    // Use a separate executor service to start the new lb, so both lbs can start concurrently.
+    _newLbExecutor.execute(() -> _newLb.start(getStartUpCallback(true,
+        mode == DualReadModeProvider.DualReadMode.NEW_LB_ONLY ? callback : null)
+    ));
+
+    _oldLb.start(getStartUpCallback(false,
+        mode == DualReadModeProvider.DualReadMode.NEW_LB_ONLY ? null : callback
+    ));
+  }
+
+  private Callback<None> getStartUpCallback(boolean isForNewLb, Callback<None> callback)
+  {
+    return new Callback<None>() {
       @Override
-      public void onError(Throwable e)
-      {
-        LOG.warn("Failed to start new load balancer. Fall back to read from old balancer only", e);
-        _isNewLbReady = false;
+      public void onError(Throwable e) {
+        LOG.warn("Failed to start {} load balancer.", isForNewLb ? "new" : "old", e);
```

Comment:
unfortunately it won't work since both new and old lbs are wrapped in a WarmUpLoadBalancer, so both of them will be shown as WarmUpLoadBalancer which is not meaningful.